### PR TITLE
Replace performance_sms/by_domain

### PR DESCRIPTION
**Pull request by nickpell**
@dannyroberts cc: @NoahCarnahan

### CODE REVIEW

File: corehq/apps/performance_sms/dbaccessors.py [hunk 1]
```diff
@@ -23,9 +21,9 @@ def by_interval(interval_keys):
...
     assert settings.UNIT_TESTING
     db = PerformanceConfiguration.get_db()
-    all_docs = [row['doc'] for row in db.view(
-        'performance_sms/by_domain',
-        reduce=False,
-        include_docs=True
-    )]
-    db.bulk_delete(all_docs)
+
+    # Since this is a unit test, number of domains is small
```
Comment (NoahCarnahan) [inline corehq/apps/performance_sms/dbaccessors.py]: Looks like this function could be called outside of a unit test. Might want to add the "unit test only" decorator (I don't recall the exact decorator name).

Comment (nickpell) [inline corehq/apps/performance_sms/dbaccessors.py]: done

**Comment by dannyroberts** [comment]
(not mergeable)

**Comment by nickpell** [comment]
@dannyroberts resolved conflicts

**Comment by dannyroberts** [comment]
not mergeable (again)

**Comment by czue** [comment]
fyi this is also only on the meta db. which means that it's both less useful and also fine to merge and reindex immediately assuming we want to kill the whole view.

**Comment by dannyroberts** [comment]
:+1: Cory's comment also applies to https://github.com/dimagi/commcare-hq/pull/9206. Fwiw, even though it's unlikely to have a great impact on db performance, it's still positive cleanup, that makes it easier to spot unnecessary views. (i.e. it's hard to tell by looking at this view that the reason it doesn't matter is because it's in an off-main db; that makes it seem like adding views like this is an acceptable thing to emulate; by getting rid of all of them in favor of a single view it makes it easier to stick to that as the only acceptable pattern.) If this were harder to do, then I'd drop it, but given the level of effort I'd say it's worthwhile.

**Comment by nickpell** [comment]
:white_check_mark:

**Comment by czue** [comment]
@dannyroberts I agree with that. though the one thing I would add is that I don't like the pattern of looping through all domains to get all of a particular type of doc to go do something with them (usually delete them in tests).

we could at least consolidate them all into a single function that takes in a list of doc types so that we could clean it up later if we want.

File: corehq/apps/performance_sms/dbaccessors.py (hunk 1 above)
Comment (dannyroberts) [inline corehq/apps/performance_sms/dbaccessors.py]: To @czue's note, I agree there's a better way to delete all `PerformanceConfiguration` docs. In fact, such a function exists! See `delete_all_docs_by_doc_type`.

Comment (dannyroberts) [inline corehq/apps/performance_sms/dbaccessors.py]: I think that would make this whole function

```python
`delete_all_docs_by_doc_type(PerformanceConfiguration.get_db(), 'PerformanceConfiguration')`
```

Comment (czue) [inline corehq/apps/performance_sms/dbaccessors.py]: :+1: nice

**Pull request merged**